pkg/streamingpromql/query.go

```diff
@@ -207,6 +207,12 @@ func (q *Query) Exec(ctx context.Context) *promql.Result {
 			return &promql.Result{Err: err}
 		}
 		defer types.SeriesMetadataSlicePool.Put(&series, q.memoryConsumptionTracker)
+		// TODO: place this as part of Put call above.
+		defer func() {
+			for _, s := range series {
+				q.memoryConsumptionTracker.DecreaseMemoryConsumptionForLabels(s.Labels)
+			}
+		}()
```

Comment:
Is this still needed? Seems like it would be covered by the hook as well.

Reply:
Remnant from rebase process. I will remove that.